src/@types/index.ts

```diff
@@ -0,0 +1,415 @@
+     */
+    handleError(errorType: KnownErrors, innerError: Error): void
+
+    /**
+     * A hook enabling providers to perform encoding of beacon data before
+     * sending it, such as `JSON.stringify()` or other serialization methods.
+     * @param testConfig The test configuration.
+     * @param data The data to be encoded.
+     */
+    encodeBeaconData(testConfig: TestConfiguration, data: Beacon.Data): string
+
+    /**
+     * Called when the Promise returned by {@link sendBeacon} is rejected.
+     * @param error An error describing the problem.
+     */
+    onSendBeaconRejected(error: Error): void
+
+    /**
+     * Called when the Promise returned by {@link sendBeacon} is resolved.
+     * @param result The Response object in case the Fetch API was used, or
+     * `void` if the Beacon API was used.
+     */
+    onSendBeaconResolved(result: SendBeaconResult): void
+
+    /**
+     * A hook enabling providers to report test results, i.e. "beaconing".
+     * @param testConfig The test configuration.
+     * @param encodedBeaconData The beacon payload returned from
+     * {@link Provider.encodeBeaconData}.
+     */
+    sendBeacon(
+        testConfig: TestConfiguration,
+        encodedBeaconData: string,
+    ): Promise<SendBeaconResult>
+}
+
+/**
+ * A function that takes an array of {@link SessionConfig} objects and returns
+ * a Promise<{@link SessionResult}>.
+ *
+ * @remarks
+ * In general, the list of {@link SessionConfig} objects passed into this
+ * function would be the result of previous calls to
+ * {@link Provider.fetchSessionConfig} for each provider used by the client.
+ *
+ * The tag owner may override the default sequencing function in order to
+ * control the order in which tests run.
+ *
+ * @param sessionConfigs List of session configurations for which to
+ * generate a list of Promise<{@link SessionResult}> objects.
+ */
+export type SessionProcessFunc = (
+    sessionConfigs: SessionConfig[],
+) => Promise<SessionResult>
+
+/**
+ * Used by the tag owner to pass settings to the client at runtime. In general,
+ * these settings affect core module behavior at the page level. For example,
+ * the site might pass different settings to the client on one page than it
+ * does on another.
+ */
+export interface ClientSettings {
+    /**
+     * The approximate delay (in milliseconds) after {@link init} is called
+     * before the RUM session should begin.
+     */
+    preConfigStartDelay?: number
+    /**
+     * An array of objects implementing {@link Provider}. When {@link init} is
+     * called, these providers will participate in the RUM session.
+     */
+    providers: Provider[]
+    /**
+     * At runtime, this function takes the list of {@link SessionConfig}
+     * objects supplied by providers participating in the RUM session
+     * and executes any tests they contain.
+     *
+     * @remarks
+     * Defaults to {@link defaultSequenceFunc}
+     */
+    sessionProcess: SessionProcessFunc
```

Comment:
If this has a default set elsewhere in the library, should it be optional?
```suggestion
    sessionProcess?: SessionProcessFunc
```

Reply:
This is a good point. Right now there is a default, but it's not in a good place...the `ClientSettingsBuilder` constructor. It seems more intuitive to have it be optional here, and then have the library supply the default downstream if not supplied.